private/buf/bufsync/syncer.go

```diff
@@ -651,16 +631,6 @@ func (s *syncer) backfillTags(
 	return nil
 }
 
-type readModuleOpts struct {
-	expectedModuleIdentity string
-}
-
-type readModuleAtOption func(*readModuleOpts)
-
-func readModuleAtWithExpectedModuleIdentity(moduleIdentity string) readModuleAtOption {
```

Comment:
All except this option lgtm, let's chat about that. We don't want any buildable module to sync to the module identity in HEAD.

Reply:
Fixed